tests/TestCase/Controller/SentencesControllerTest.php

```diff
@@ -295,12 +295,44 @@ public function testPaginateRedirectsPageOutOfBoundsToLastPage() {
                 'user_id' => 1,
             ];
         }
-        $sentences = TableRegistry::get('Sentences');
+        $sentences = TableRegistry::getTableLocator()->get('Sentences');
         $entities = $sentences->newEntities($newSentences);
-        $result = $sentences->saveMany($entities);
+        $sentences->saveMany($entities);
+
+        $nbSentences = count($sentences->find()->where(['user_id' => $userId])->all());
+        $lastPage = ceil($nbSentences / $defaultNbPerPage);
 
         $this->get("/eng/sentences/of_user/$user?page=9999999");
+        $this->assertRedirect("/eng/sentences/of_user/$user?page=$lastPage");
+    }
+
+    public function testPaginateRedirectsPageOutOfBoundsToLastPage_withUserSetting() {
+        $user = 'kazuki';
+        $userId = 7;
+        $users = TableRegistry::getTableLocator()->get('Users');
+        $NbPerPageSetting = $users->getSettings($userId)['settings']['sentences_per_page'];
+        $newSentences = array();
+        for ($i = 1; $i <= $NbPerPageSetting + 1; $i++) {
+            $newSentences[] = [
+                'lang' => 'eng',
+                'text' => "Ay ay ay $i.",
+                'user_id' => $userId,
+            ];
+            $newSentences[] = [
+                'lang' => 'eng',
+                'text' => "Oy oy oy $i.",
+                'user_id' => 1,
+            ];
+        }
+        $sentences = TableRegistry::getTableLocator()->get('Sentences');
+        $entities = $sentences->newEntities($newSentences);
+        $sentences->saveMany($entities);
```

Comment:
How about refactoring this code into a private method like addManySentencesAsUser()?